src/libraries/Microsoft.Extensions.Logging.Console/src/SimpleConsoleFormatter.cs

```diff
@@ -155,10 +157,50 @@ private static string GetLogLevelString(LogLevel logLevel)
             };
         }
 
+        /// <summary>Get whether to emit ANSI color codes.</summary>
+        private static bool EmitAnsiColorCodes
+        {
+            get
+            {
+                // The flag starts at -1.  If it's no longer -1, it's 0 or 1 to represent false or true.
+                int emitAnsiColorCodes = s_emitAnsiColorCodes;
```

Comment:
I assume SimpleConsoleFormatter in a single instance for the host lifetime.
Then it is more interesting to cache `disableColors` in an instance field than `s_emitAnsiColorCodes` in a static.

Reply:
I ended up moving the logic to Common folder, used by both `ConsolePal.Unix.cs` and here.